src/main.cpp

```diff
@@ -34,36 +47,74 @@
 #        include "clientdlg.h"
 #    endif
 #endif
-#include "settings.h"
+
+#include "rpcserver.h"
+#include "serverrpc.h"
 #ifndef SERVER_ONLY
+#    include "clientrpc.h"
 #    include "testbench.h"
 #endif
-#include "util.h"
+
 #ifdef ANDROID
 #    include <QtAndroidExtras/QtAndroid>
 #endif
+
 #if defined( Q_OS_MACX )
 #    include "mac/activity.h"
 extern void qt_set_sequence_auto_mnemonic ( bool bEnable );
 #endif
-#include <memory>
-#include "rpcserver.h"
-#include "serverrpc.h"
-#ifndef SERVER_ONLY
-#    include "clientrpc.h"
-#endif
+
+// Global App Name *************************************************************
+
+static QString strAppName = APP_NAME; // Will be appended by " - ClientName" or "Server - ServerName" if those names are defined
+
+const QString GetAppName() { return strAppName; }
+
+// Helpers *********************************************************************
+
+const QString getServerNameFromInfo ( const QString strServerInfo )
+{
+    if ( !strServerInfo.isEmpty() )
+    {
+        QStringList servInfoParams = strServerInfo.split ( ";" );
+        if ( servInfoParams.count() > 0 )
+        {
+            return servInfoParams[0];
+        }
+    }
+
+    return QString();
+}
 
 // Implementation **************************************************************
 
 int main ( int argc, char** argv )
 {
+    int exit_code = 0;
 
 #if defined( Q_OS_MACX )
     // Mnemonic keys are default disabled in Qt for MacOS. The following function enables them.
     // Qt will not show these with underline characters in the GUI on MacOS. (#1873)
     qt_set_sequence_auto_mnemonic ( true );
 #endif
 
+#ifdef HEADLESS
```

Comment:
Remove this, and just replace the existing pointer declarations with `QScopedPtr` (or `std::unique_ptr`), keeping them in the same places as the original code. There is no need for the `#define`s - `pApplication` can just be a pointer to a `QCoreApplication`, as `QCoreApplication` has (as usual) a virtual destructor.